### PR TITLE
Hide table of contents when necessary

### PR DESCRIPTION
**Pull request by mic4ael**

### CODE REVIEW

**Comment by ThiefMaster** [comment]
If a commit fixes an issue, can to add `closes #xxx` to the commit message?

File: indico/legacy/pdfinterface/base.py [hunk 1]
```diff
@@ -621,7 +620,8 @@ def __init__(self, story=None, pagesize='A4', fontsize='normal', firstPageNumber
         self._fileDummy = FileDummy()
 
         self._doc = DocTemplateWithTOC(self._indexedFlowable, self._fileDummy, firstPageNumber=firstPageNumber,
-                                       pagesize=PDFSizes().PDFpagesizes[pagesize])
+                                       pagesize=PDFSizes().PDFpagesizes[pagesize],
+                                       include_table_of_contents=include_table_of_contents)
```
Comment (pferreir) [inline indico/legacy/pdfinterface/base.py]: `include_toc`, maybe?

File: indico/legacy/pdfinterface/base.py [hunk 2]
```diff
@@ -503,13 +503,14 @@ def _doNothing(canvas, doc):
...
 
-    def __init__(self, indexedFlowable, filename, firstPageNumber = 1, **kw ):
+    def __init__(self, indexedFlowable, filename, firstPageNumber=1, **kw):
         """toc is the TableOfContents object
         indexedFlowale is a dictionnary with flowables as key and a dictionnary as value.
             the sub-dictionnary have two key:
                 text: the text which will br print in the table
                 level: the level of the entry( modifying the indentation and the police
         """
+        self.include_table_of_contents = kw.get('include_table_of_contents', False)
```
Comment (pferreir) [inline indico/legacy/pdfinterface/base.py]: Why not add this as a kwarg of the constructor?

File: indico/legacy/pdfinterface/base.py (hunk 1 above)
Comment (mic4ael) [reply indico/legacy/pdfinterface/base.py]: ok, I don't mind changing it

File: indico/legacy/pdfinterface/base.py (hunk 2 above)
Comment (mic4ael) [reply indico/legacy/pdfinterface/base.py]: don't mind changing it

Comment (mic4ael) [reply indico/legacy/pdfinterface/base.py]: changed

File: indico/legacy/pdfinterface/base.py [hunk 3]
```diff
@@ -503,13 +503,14 @@ def _doNothing(canvas, doc):
 
 class DocTemplateWithTOC(SimpleDocTemplate):
 
-    def __init__(self, indexedFlowable, filename, firstPageNumber = 1, **kw ):
+    def __init__(self, indexedFlowable, filename, firstPageNumber=1, include_toc=False, **kw):
```
Comment (ThiefMaster) [inline indico/legacy/pdfinterface/base.py]: fyi, this class is used in other places  where the TOC is now disabled too - so this argument should be True by default (i'm changing it, no need to update the PR)

**Pull request merged**